pkg/reconciler/reconciler.go

```diff
@@ -561,10 +568,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 			return ctrl.Result{}, err
 		}
 
+	case statedNeedsRollback:
+		{
+			if err := r.doRollback(actionClient, &u, obj); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
```

Comment:
Do we need the block?
```suggestion
	case statedNeedsRollback:
		if err := r.doRollback(actionClient, &u, obj); err != nil {
			return ctrl.Result{}, err
		}
```